File: api_server/models/relations.py
from . import database, db


post_catalog = db.Table(
    'post_catalogs',
    db.Column('post_id', db.ForeignKey('post.post_id')),
    db.Column('catalog_id', db.ForeignKey('catalog.catalog_id'))
)

post_tag = db.Table(
    'post_tags',
    db.Column('post_id', db.ForeignKey('post.post_id')),
    db.Column('tag_id', db.ForeignKey('tag.tag_id'))
)

File: RoseliaCore/Database/Models/Relations.fs
namespace RoseliaBlog.RoseliaCore.Database.Models.Relations

open System.ComponentModel.DataAnnotations.Schema
open RoseliaBlog.RoseliaCore.Database.Models

[<CLIMutable>]
[<Table("post_catalogs")>]
type PostCatalog = {
    [<Column("post_id")>]
    PostId: int
    Post: Post
    
    [<Column("catalog_id")>]
    CatalogId: int
    Catalog: Catalog
}

[<CLIMutable>]
[<Table("post_tags")>]
type PostTag = {
    [<Column("post_id")>]
    PostId: int
    Post: Post
    
    [<Column("tag_id")>]
    TagId: int
    Tag: Tag
}
